zebra-rpc/src/methods.rs

```diff
@@ -1133,73 +1062,54 @@ where
                 _ => unreachable!("unmatched response to a block request"),
             };
 
-            let hash = hash_or_height.hash().unwrap_or_else(|| block.hash());
-            let hash_or_height = hash.into();
-
-            // Fetch the Sapling & Orchard treestates referenced by
-            // [`hash_or_height`] from the state.
-
-            let sapling_request = zebra_state::ReadRequest::SaplingTree(hash_or_height);
-            let sapling_response = state
-                .ready()
-                .and_then(|service| service.call(sapling_request))
-                .await
-                .map_err(|error| Error {
-                    code: ErrorCode::ServerError(0),
-                    message: error.to_string(),
-                    data: None,
-                })?;
-
-            let orchard_request = zebra_state::ReadRequest::OrchardTree(hash_or_height);
-            let orchard_response = state
-                .ready()
-                .and_then(|service| service.call(orchard_request))
-                .await
-                .map_err(|error| Error {
-                    code: ErrorCode::ServerError(0),
-                    message: error.to_string(),
-                    data: None,
-                })?;
+            let hash = hash_or_height
+                .hash_or_else(|_| Some(block.hash()))
+                .expect("block hash");
```

Comment:
Optional/Nitpick: Could we keep using `unwrap_or_else()` here?

```suggestion
            let hash = hash_or_height.hash().unwrap_or_else(|| block.hash());
```